src/Corax/IndexSearcher/IndexSearcher.cs

```diff
@@ -52,37 +53,42 @@ public sealed unsafe partial class IndexSearcher : IDisposable
     private readonly Tree _metadataTree;
     private readonly Tree _fieldsTree;
 
+    private readonly bool _documentsAreBoosted;
+    public bool DocumentsAreBoosted => _documentsAreBoosted;
+
     // The reason why we want to have the transaction open for us is so that we avoid having
     // to explicitly provide the index searcher with opening semantics and also every new
     // searcher becomes essentially a unit of work which makes reusing assets tracking more explicit.
-    public IndexSearcher(StorageEnvironment environment, IndexFieldsMapping fieldsMapping = null) : this(fieldsMapping)
+    public IndexSearcher(StorageEnvironment environment, IndexFieldsMapping fieldsMapping = null) : this(fieldsMapping, documentsAreBoosted: false)
     {
         _ownsTransaction = true;
         _transaction = environment.ReadTransaction();
         _fieldsTree = _transaction.ReadTree(Constants.IndexWriter.FieldsSlice);
         _metadataTree = _transaction.ReadTree(Constants.IndexMetadataSlice);
     }
 
-    public IndexSearcher(Transaction tx, IndexFieldsMapping fieldsMapping = null) : this(fieldsMapping)
+    public IndexSearcher(Transaction tx, IndexFieldsMapping fieldsMapping = null, bool documentsAreBoosted = false) : this(fieldsMapping, documentsAreBoosted)
     {
         _ownsTransaction = false;
         _transaction = tx;
         _fieldsTree = _transaction.ReadTree(Constants.IndexWriter.FieldsSlice);
         _metadataTree = _transaction.ReadTree(Constants.IndexMetadataSlice);
     }
 
-    private IndexSearcher(IndexFieldsMapping fieldsMapping)
+    private IndexSearcher(IndexFieldsMapping fieldsMapping, bool documentsAreBoosted)
     {
         if (fieldsMapping is null)
         {
             _ownsIndexMapping = true;
-            using var builder =  IndexFieldsMappingBuilder.CreateForReader();
+            using var builder = IndexFieldsMappingBuilder.CreateForReader();
             _fieldMapping = builder.Build();
         }
         else
         {
             _fieldMapping = fieldsMapping;
         }
+
+        _documentsAreBoosted = documentsAreBoosted;
```

Comment:
```suggestion
        _documentsAreBoosted = GetDocumentBoostTree().Count;
```

That would be better?

Reply:
Agree